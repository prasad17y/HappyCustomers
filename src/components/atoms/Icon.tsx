import React from 'react';
import {
  Image,
  StyleSheet,
  ImageStyle,
  StyleProp,
  ImageSourcePropType,
} from 'react-native';

interface IconProps {
  source: ImageSourcePropType;
  style?: StyleProp<ImageStyle>;
}

const Icon: React.FC<IconProps> = ({source, style}) => {
  return <Image source={source} style={[styles.icon, style]} />;
};

const styles = StyleSheet.create({
  icon: {
    width: 24,
    height: 24,
  },
});

export default React.memo(Icon);
