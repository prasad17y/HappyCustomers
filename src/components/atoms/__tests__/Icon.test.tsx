import React from 'react';
import {render} from '@testing-library/react-native';
import Icon from '../Icon';

describe('Icon', () => {
  it('should render correctly and match the snapshot', () => {
    const tree = render(
      <Icon source={require('../../../assets/search.png')} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
