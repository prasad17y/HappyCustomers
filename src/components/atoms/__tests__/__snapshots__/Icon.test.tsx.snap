// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly and match the snapshot 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/search.png",
    }
  }
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      undefined,
    ]
  }
/>
`;
